### ISSUE TITLE
[REQ] Workout history and start of previous trainings

### ISSUE THREAD

**Issue by blautob** [cagnulein/qdomyos-zwift#2383]
Hi Roberto

i have an idea for a feature:

In the charts menu it would be great to flip through all previous workouts (see my mockup)

additionaly for peloton for example it would be fantastic to start the already saved and downloaded training again out of this screen.

what do you think?

![IMG_4500](https://github.com/cagnulein/qdomyos-zwift/assets/24385671/09014714-d10c-4c94-b31e-680fd995c3a1)

**Comment by cagnulein** [comment]
yeah i already started something similar but it also requires a database to
store the workouts.
thanks for the idea

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 11 giu 2024 alle 12:03 blautob ***@***.***> ha
scritto:

**Comment by blautob** [comment]
great! as always you are ahead of it already.

do you plan to include the restart of the training plan feature as well. because i think that would really be a killer feature

if you need somebody for testing, happy to so on ios testflight

**Comment by cagnulein** [comment]
Hah Ok yes got it. Yes should be easy to do once the history works

**Comment by blautob** [comment]
Hi @cagnulein 
thanks for letting me test the beta feature of the workout history.
i like it quite much
Some initial thougts/improvements:
i like the way e.g. peloton motivates you in termes of "you had a 3 day streak of training". maybe even a small calenda ron top which shows trainings would be cool.

what i would love is the connection to peloton rides that i took in the respective workout.
even better clickable link on the class and possibility to load the workout from the training program section.


P.s. i had to uninstall the version unfortunately, as the beta version broke the tiles peloton target cadence and peloton target resisitance for me unfortunatley (always showing 0 and no range)

**Comment by cagnulein** [comment]
> P.s. i had to uninstall the version unfortunately, as the beta version broke the tiles peloton target cadence and peloton target resisitance for me unfortunatley (always showing 0 and no range)

yeah i'm building the workout history version with this patch as well



> what i would love is the connection to peloton rides that i took in the respective workout.
> even better clickable link on the class and possibility to load the workout from the training program section.

could you please describe it more?



> i like the way e.g. peloton motivates you in termes of "you had a 3 day streak of training". maybe even a small calenda ron top which shows trainings would be cool.

ok i will think about, maybe a small banner in the bottom with the current streak?

**Comment by blautob** [comment]
> could you please describe it more?

for example i start a peloton ride : "30 minutes motown". Then the ride in the history is named "30 minutes motown on 04.08.2025). then i click on this ride in the ride history and see all the great charts and stats. on the top/bottom i can click on "30 minutes motown" and it transfers me to QZ's "open train program" > Ride> Instructor> "30 minutes motown" 
i can start the ride again...
additionaly there is a link to Https:/members.onepeloton.com/classes/"30 minutes motown"

**Comment by cagnulein** [comment]
Got it i will think about!
I have ready the new version if you want to try it

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 4 ago 2025 alle 15:41 blautob ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
i added calendar and the streak banner! in the 2.20.7

**Comment by blautob** [comment]
Great. Cant see the version in testflight yet. Looking forward 👍

**Comment by cagnulein** [comment]
yes i didn't publish it yet because i'm adding also the other suggestions
that you mentioned before!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 5 ago 2025 alle ore 15:45 blautob ***@***.***>
ha scritto:

**Comment by blautob** [comment]
Fantastic. Looking forward!

**Comment by cagnulein** [comment]
@blautob i just sent you the new version! It should have everything you asked. Consider that the peloton tag and workout and url will be visible for the new workout done with this version and future ones (before i wasn't saving the correlation between the peloton workout, url and xml). in case i can try to add a lookup function but for now i guess it's enough for testing it!

**Comment by cagnulein** [comment]
hah also it will be good to share some screenshot on the community on facebook or reddit to show the new features!

**Comment by blautob** [comment]
WOW. This looks even better than i imagined. 
I‘ll do a ride later and check the peloton part. 
Fantastic work again.

**Comment by cagnulein** [comment]
Glad to hear it!
I worked hard on this following your advice!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 6 ago 2025 alle ore 13:49 blautob ***@***.***>
ha scritto:

**Comment by blautob** [comment]
Very small improvement idea
Put the qz or peloton logo badge on the left side above the bike emoji as longer texts overwrite it

**Comment by cagnulein** [comment]
Sure!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 6 ago 2025 alle 19:09 blautob ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
<img width="488" height="783" alt="Image" src="https://github.com/user-attachments/assets/ba8d01b0-604c-42df-b8f6-a5a66789715c" />

done is it ok?

**Comment by blautob** [comment]
Perfect

**Comment by cagnulein** [comment]
thanks, next build will have this!
did you test also the link and the training program from peloton in that
screen?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 7 ago 2025 alle ore 09:36 blautob ***@***.***>
ha scritto:

**Comment by blautob** [comment]
Yes. Both worked like a charm

**Comment by cagnulein** [comment]
unfortunately i can't put on the left because on some mobiles it conflicts with the title. i will put scrollable title or multiline title so it will not overlap with the peloton tag

**Comment by blautob** [comment]
Scrolling text looks nice in the new version. However the button to the gpx file is missing for yesterdays rides or do i have to redo a ride on the new version. 

I just thought about a search/filter field just like in „workouts“. What do you think?

**Comment by cagnulein** [comment]
gpx file? what do you mean for peloton? do you mean xml? you have to do
with the version that i sent yesterday

rework the training program view could be a thing. i will think about it

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 7 ago 2025 alle 14:07 blautob ***@***.***> ha
scritto:

**Comment by blautob** [comment]
Yes sorry i meant the xml file. I could see it in yesterdays version on yesterdays rides. But not visible todays version on yesterdays ride. I will do a ride today on todays version

**Comment by cagnulein** [comment]
that's strange, do you see the xml in the training program section?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 7 ago 2025 alle ore 15:14 blautob ***@***.***>
ha scritto:

**Comment by blautob** [comment]
Oh no. 😿 icant see any xml (other than the default) in train program

**Comment by blautob** [comment]
Sorry my bad They are still there in my Doxuments app>“training“ and in open teain program in QZ

**Comment by cagnulein** [comment]
ok so you simply don't have the button in the workout history ? even for
the one that you did with the today and yesterday version?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 7 ago 2025 alle ore 15:38 blautob ***@***.***>
ha scritto:

**Comment by blautob** [comment]
Yes no button anymore

**Comment by cagnulein** [comment]
<img width="508" height="496" alt="Image" src="https://github.com/user-attachments/assets/263a3a2c-3b54-4ca0-a68c-dab46e969912" />

strange i still have it. do you have a debug log showing me the issue on the workout history?

**Comment by blautob** [comment]
Can confirm that the tbutton is now back for new rides!

**Comment by cagnulein** [comment]
Amazing!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 14 ago 2025 alle 08:13 blautob ***@***.***> ha
scritto:
